File: BurninWiper/ColorActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using Android.App;
using Android.Content;
using Android.Graphics.Drawables;
using Android.OS;
using Android.Preferences;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace BurninWiper
{
	[Activity(Label = "ColorActivity")]
	public class ColorActivity : Activity
	{
		private LinearLayout ColorLinearLayout;
		protected override void OnCreate(Bundle savedInstanceState)
		{
			base.OnCreate(savedInstanceState);
			SetContentView(Resource.Layout.color_main);

			shutdownTimer.Elapsed += (_, __) => Stop();
			ColorLinearLayout = FindViewById<LinearLayout>(Resource.Id.ColorLinearLayout);
			ColorLinearLayout.Click += (_, __) => Finish();
			Start();
		}

		//For turning off automatically
		private readonly System.Timers.Timer shutdownTimer = new System.Timers.Timer();
		private ColorDrawable[] CurrentColors;

		public int ShutdownTimerIndex
		{
			set
			{
				int MinutesToMs(int s) => s * 60 * 1000;
				if (value == 0) shutdownTimer.Interval = int.MaxValue;
				else if (value == 1) shutdownTimer.Interval = MinutesToMs(5);
				else if (value == 2) shutdownTimer.Interval = MinutesToMs(10);
				else if (value == 3) shutdownTimer.Interval = MinutesToMs(20);
				else if (value == 4) shutdownTimer.Interval = MinutesToMs(30);
				else if (value == 5) shutdownTimer.Interval = MinutesToMs(60);
				else if (value == 6) shutdownTimer.Interval = MinutesToMs(120);
				else if (value == 7) shutdownTimer.Interval = MinutesToMs(180);
				else shutdownTimer.Interval = int.MaxValue;
			}
		}
		public int ColorMode
		{
			set => CurrentColors = Colors.GetProperArray(value);
		}

		public Thread ColorChanger { get; private set; }

		private void ColorRotation()
		{
			while (true)
			{
				for (int i = 0; i < CurrentColors.Length; i++)
				{
					RunOnUiThread(() => ColorLinearLayout.Background = CurrentColors[i]);
					Thread.Sleep(1500);
				}
			}
		}

		private void MakeAppImmersive()
		{
			//Numbers are representing SystemUiFlags enumeration (Immersive | FullScreen | HideNavigation);
			Window.DecorView.SystemUiVisibility = (StatusBarVisibility)(2048 | 4 | 2);
		}
		public void Stop()
		{
			shutdownTimer.Stop();
			ColorChanger?.Abort();
			Finish();
		}


		public void Start()
		{
			Window.SetFlags(WindowManagerFlags.KeepScreenOn, WindowManagerFlags.KeepScreenOn);
			ApplyPreferences();
			MakeAppImmersive();
			StartThread();
			shutdownTimer.Start();
		}

		private void StartThread()
		{
			ColorChanger = new Thread(ColorRotation);
			ColorChanger.Start();
		}

		public void ShowGray()
		{
			MakeAppImmersive();
			ColorLinearLayout.Background = Colors.Gray;
		}
		private void ApplyPreferences()
		{
			var prefs = PreferenceManager.GetDefaultSharedPreferences(this);
			if (prefs.GetBoolean("IsChecking", false))
			{
				var edit = prefs.Edit();
				edit.PutBoolean("IsChecking", false);
				edit.Commit();
				ShutdownTimerIndex = 0;
				ColorMode = 4;
				return;
			}
			ColorMode = prefs.GetInt("ModeIndex", 0);
			ShutdownTimerIndex = prefs.GetInt("TimerIndex", 0);
		}
	}
}

File: app/src/main/java/soheilkd/BurninWiper/ColorActivity.kt
package soheilkd.BurninWiper

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.preference.PreferenceManager
import android.view.View
import kotlinx.android.synthetic.main.activity_color.*
import java.util.*
import kotlin.concurrent.timerTask

class ColorActivity : AppCompatActivity() {
	private var currentColors: Array<ColorDrawable> = arrayOf()
	private var colorThread = Thread(Runnable {
		while(true){
			for(color in currentColors){
				runOnUiThread { ColorFrameLayout.background = color }
				Thread.sleep(1250)
			}
		}
	})
	 private fun start(){
		window.setFlags(128, 128)
		applyPreferences()
		makeAppImmersive()
		 colorThread.start()
	}
	private fun makeAppImmersive(){
		ColorFrameLayout.systemUiVisibility =
			View.SYSTEM_UI_FLAG_LOW_PROFILE or
					View.SYSTEM_UI_FLAG_FULLSCREEN or
					View.SYSTEM_UI_FLAG_LAYOUT_STABLE or
					View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY or
					View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION or
					View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
	}
	private var shutdownTimer = Timer()
	private fun setShutdownTimer(index: Int){
		val timerTask = timerTask {
			finish()
		}
		fun minToMs(m: Int): Long {return  m*60L*1000}
		when(index) {
			1 -> shutdownTimer.schedule(timerTask, minToMs(5))
			2 -> shutdownTimer.schedule(timerTask, minToMs(10))
			3 -> shutdownTimer.schedule(timerTask, minToMs(20))
			4 -> shutdownTimer.schedule(timerTask, minToMs(30))
			5 -> shutdownTimer.schedule(timerTask, minToMs(60))
			6 -> shutdownTimer.schedule(timerTask, minToMs(120))
			7 -> shutdownTimer.schedule(timerTask, minToMs(180))
		}
	}

	private fun getColorArray(id: Int): Array<ColorDrawable>{
		val white by lazy { ColorDrawable(Color.WHITE) }
		val red by lazy { ColorDrawable(Color.RED) }
		val green by lazy { ColorDrawable(Color.GREEN) }
		val blue by lazy { ColorDrawable(Color.BLUE) }
		val black by lazy { ColorDrawable(Color.BLACK) }
		val gray by lazy { ColorDrawable(Color.DKGRAY) }
		return when (id){
			1 -> arrayOf(white,red,green,blue,black)
			2 -> arrayOf(white,black)
			3 -> arrayOf(white)
			else -> arrayOf(gray)
		}
	}
	private fun setColors(id: Int) {
	currentColors = getColorArray(id)
	}
	private fun applyPreferences(){
		val prefs = PreferenceManager.getDefaultSharedPreferences(this)
		if (prefs.getBoolean("IsChecking", false)){
			val edit = prefs.edit()
			edit.putBoolean("IsChecking", false)
			edit.apply()
			setColors(4)
			return
		}
		setColors(prefs.getInt("ModeIndex", 0))
		setShutdownTimer(prefs.getInt("TimerIndex" , 0))
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		setContentView(R.layout.activity_color)
		supportActionBar?.setDisplayHomeAsUpEnabled(true)

		ColorFrameLayout.setOnTouchListener{ _, _ ->
			finish()
			true
		}
		start()
	}
/*
	private fun hide() {
		// Hide UI first
		supportActionBar?.hide()
		fullscreen_content_controls.visibility = View.GONE
		mVisible = false

		// Schedule a runnable to remove the status and navigation bar after a delay
		mHideHandler.removeCallbacks(mShowPart2Runnable)
		mHideHandler.postDelayed(mHidePart2Runnable, UI_ANIMATION_DELAY.toLong())
	}*/
}
